### DISCUSSION TITLE
How to implement search autocomplete?

### DISCUSSION THREAD

**Discussion by naveencom** [unpoly/unpoly#479] [Q&A] [answered]
I want to implement a search autocomplete similar like Google search... Please check the attached screenshot. 

![google-search](https://user-images.githubusercontent.com/57630189/235596688-aad840cf-357c-40f4-9f52-ecb7407973cd.png)

As as beginner to Unpoy js, I do not know how to do this.

Can someone please help me?

**Comment by triskweline** [comment, marked as answer]
I would make a form like this:

```html
<form method="get" action="/search">
  <input type="search" name="query" up-validate=".suggestions" up-watch-event="input">
  <div class="suggestions"></div>
</form>
```

When your `/search` endpoint gets a request with an [`X-Up-Validate`](https://unpoly.com/X-Up-Validate) you should render a `<div class="suggestions">` element with your suggestions.

See documentation for [Dependent fields](https://unpoly.com/dependent-fields).

**Reply by naveencom** [reply]
Thanks for the response.
